File: Record.py
class Record:
    def __init__(self, UUID, type, creator_id, receiver_id, receiver_name, confirmed, receiver_paid, creator_paid, active, date_created, amount, note):
        self.UUID = UUID
        self.type = type   # 0 = creator borrows from receiver, 1 = creator lending to receiver
        self.creator_id = creator_id
        self.receiver_id = receiver_id
        self.receiver_name = receiver_name   # in case receiver doesnt have an account, can use name (expansion feature)
        self.confirmed = confirmed   # receiver confirms record as being correct
        self.receiver_paid = receiver_paid   # receiver confirms record as being paid
        self.creator_paid = creator_paid   # creator confirms record as being paid
        self.active = active   # if already marked as paid by both, record becomes inactive
        self.date_created = date_created
        self.amount = amount
        self.note = note

File: LJ Borrow/LJ Borrow/Record.swift
//
//  Record.swift
//  LJ Borrow
//
//  Created by Abi on 25/06/24.
//

import Foundation

struct Record: Identifiable, Codable {
    var id: String
    var type: Int
    var creator_id: String
    var receiver_id: String
    var associated_name: String
    var confirmed: Int
    var receiver_paid: Int
    var creator_paid: Int
    var active: Int
    var date_created: String
    var amount: Double
    var note: String
    
    var typeDescription: String {
        (type == 1 && creator_id == UserSession.shared.userUUID || type == 0 && receiver_id == UserSession.shared.userUUID) ? "Lent" : "Borrowed"
    }
    
    var adjustedAmount: Double {
        amount / 100
    }
}
